### DISCUSSION TITLE
Lesson 13: Wallet Balance Does Not Update

### DISCUSSION THREAD

**Discussion by Niferu** [smartcontractkit/full-blockchain-solidity-course-py#1728] [Q&A] [answered]
Hello,

As in title my wallet balances does not update on switching token, there is 0 for no matter what i click. I have check `brownie-config.yaml` token addresses and DappToken address from `map.json` and those match with my MetaMask account where I have blance for those tokens. I have also copied `Main.tsx`, `YourWallet.tsx`, `WalletBalance.tsx` and `BalanceMsg.tsx` from repository to see if I have any typo in code but it still shows 0 everywhere, so I believe there might be issue with mapping but have no idea how to fix it.

![image](https://user-images.githubusercontent.com/110537615/193124378-b1da5ddc-320e-4400-bcfc-6775614cb266.png)

Not sure if it can be because of that `Failed to parse source map from` warning, below is screen from compiling node:

![image](https://user-images.githubusercontent.com/110537615/193124897-96b50dbb-f4fd-4c91-815f-45019af88298.png)

Any help will be appreciated, thanks!

**Comment by cromewar** [comment, marked as answer]
Hello @Neftyr I made some research on this and it seems the problem is due to the changes were made on the frameworks since the video was published. I'm not quite sure how this could be solved, did you find any work around?

Also remember lesson 13 was archived for that exact same reason.

**Reply by Niferu** [reply]
Hello @cromewar 
Thank you for reply, I was week off so had no chance to find workaround, but I'm totally green into frontend anyway. In this case i will just probably skip this part.

**Reply by Niferu** [reply]
Hello @cromewar 
I think this issue is caused by below:

![image](https://user-images.githubusercontent.com/110537615/195097802-5521c776-05a4-4630-baa3-7a6ca3a14f33.png)

My console keeps spamming that `supportedChain` is deprecated (so in this case `chainId` isnt passing value, which is address from map.json in this case?) but I have no idea how to fix it...

**Reply by Niferu** [reply]
I have actually managed to fix errors by changing `App.tsx` into below:

```tsx
import { DAppProvider, Goerli } from "@usedapp/core"
import { Header } from "./components/Header"
import { Container } from '@material-ui/core';
import { Main } from "./components/Main"

function App() {
  return (
    <DAppProvider config={{
      // 1337 is for ganache local testnet
      networks: [Goerli]
    }}>
      <Header />
      <Container maxWidth = "md">
          <Main />
      </Container>
    </DAppProvider>
  )
}

export default App;
```

But unfortunately values for tokens are still 0 and console looks as below now:

![image](https://user-images.githubusercontent.com/110537615/195108945-37997f48-e5db-4c99-b19c-c5935affea8a.png)

So maybe there is problem with that `DevTools failed to load source map:` thing, but I assume it is just browser warning and it does nothing with functionality of code.

I believe that `chainId` in `Main.tsx` doesnt work as it is not working for `App.tsx` and thats why console doesnt read correct network and address of token, but have no idea how to read network and account associated with MetaMask in other way in `Main.tsx`...

Below output is from `console.log(chainId)` and `console.log(networkName)` from `Main.tsx`.

![image](https://user-images.githubusercontent.com/110537615/195116943-221e1bff-2ca6-4191-b5bc-f99b7b64c562.png)

**Reply by cromewar** [reply]
yeah it seems the chainId returns undefined, are you testing this with goerli?

**Reply by Niferu** [reply]
Hi @cromewar 

Yes I'm working on it with Goerli

**Reply by cromewar** [reply]
Well I tried to reproduce this, but I encounter myself different problems not related to this one, again please skip this lesson is too outdated.

**Reply by Niferu** [reply]
Hi @cromewar 
Thank you for checking this. I have already finished whole course and will leave this issue. Thanks a lot for your support here. I look forward to get on my own projects soon, but will keep an eye on this discussion, so maybe I will be able to help someone too :)

**Reply by cromewar** [reply]
I highly recommend you to check some lessons on the JS version.

**Reply by Niferu** [reply]
Hello @cromewar 
Thanks for advice I was considering that too, but now I will definitely check that out! Thanks a lot mate!
